File: python3/LeetCode/allPathsSourceTarget.py
from typing import List


class Solution:
    def allPathsSourceTarget(self, graph:List[List[int]]) -> List[List[int]]:
        res = list()
        track = list()
        track.append(0)

        def dfs(src:int):
            if src == len(graph)-1:
                res.append(track[:])
                return
            for gra in graph[src]:
                track.append(gra)
                dfs(gra)
                track.pop()

        dfs(0)
        return res

File: go/LeetCode/allPathsSourceTarget.go
package main

func main() {
	
}

func allPathsSourceTarget(graph [][]int) [][]int {
	res := make([][]int, 0)
	track := make([]int, 0)
	track = append(track, 0)
	var dfs func(int)
	dfs = func(src int) {
		if src == len(graph)-1 {
			res= append(res, append([]int(nil), track...))
		}
		for _,gra := range graph[src] {
			track = append(track, gra)
			dfs(gra)
			track = track[0:len(track)-1]
		}
	}
	dfs(0)
	return res
}